libs/oracledb/langchain_oracledb/vectorstores/oraclevs.py

```diff
@@ -0,0 +1,1929 @@
+        documents = await self.amax_marginal_relevance_search_by_vector(
+            embedding,
+            k=k,
+            fetch_k=fetch_k,
+            lambda_mult=lambda_mult,
+            filter=filter,
+            **kwargs,
+        )
+        return documents
+
+    @_handle_exceptions
+    def delete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> None:
+        """Delete by vector IDs.
+        Args:
+          self: An instance of the class
+          ids: List of ids to delete.
+          **kwargs
+        """
+
+        ddl, bind_vars = _get_delete_ddl(self.table_name, ids)
+
+        connection = _get_connection(self.client)
+        if connection is None:
+            raise ValueError("Failed to acquire a connection.")
+        with connection.cursor() as cursor:
+            cursor.execute(ddl, bind_vars)
+            connection.commit()
+
+    @_handle_exceptions
+    async def adelete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> None:
+        """Delete by vector IDs.
+        Args:
+          self: An instance of the class
+          ids: List of ids to delete.
+          **kwargs
+        """
+
+        ddl, bind_vars = _get_delete_ddl(self.table_name, ids)
+
+        async def context(connection: Any) -> None:
+            with connection.cursor() as cursor:
+                await cursor.execute(ddl, bind_vars)
+                await connection.commit()
+
+        await _handle_context(self.client, context)
+
+    @classmethod
+    def _from_texts_helper(
+        cls: Type[OracleVS],
+        texts: Iterable[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> Tuple[Any, str, DistanceStrategy, str, Dict]:
+        client: Any = kwargs.get("client", None)
+        if client is None:
+            raise ValueError("client parameter is required...")
+
+        params = kwargs.get("params", {})
+
+        table_name = str(kwargs.get("table_name", "langchain"))
+
+        distance_strategy = cast(
+            DistanceStrategy, kwargs.get("distance_strategy", None)
+        )
+        if not isinstance(distance_strategy, DistanceStrategy):
+            raise TypeError(
+                f"Expected DistanceStrategy got " f"{type(distance_strategy).__name__} "
+            )
+
+        query = kwargs.get("query", "What is a Oracle database")
+
+        return client, table_name, distance_strategy, query, params
+
+    @classmethod
+    @_handle_exceptions
+    def from_texts(
+        cls: Type[OracleVS],
+        texts: Iterable[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> OracleVS:
+        (
+            client,
+            table_name,
+            distance_strategy,
+            query,
+            params,
+        ) = OracleVS._from_texts_helper(texts, embedding, metadatas, **kwargs)
+
+        drop_table_purge(client, table_name)
```

Comment:
Can you please clarify my understanding ... I have seen in javascript that user calls from_texts equavelent in batches. Like for a pdf, if its divided into 1k chunks, this function is called with 250 chunks in each call. But this drop_table_purge was removing the older 250 chunks, i had to comment for my testing. Should this be done based on some options ,.. ? other vector stores do same way?